test/js/bun/shell/bunshell.test.ts

```diff
@@ -831,3 +832,61 @@ function sentinelByte(buf: Uint8Array): number {
   }
   throw new Error("No sentinel byte");
 }
+
+test("shell as script runner propagates env vars: bun run <foo>", async () => {
```

Comment:
this should be condensed into a `test.each` for the `bun run` permutation and the non-bun run permutation